### ISSUE TITLE
Unit tests fail when run from the bridge

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#1345]
The following file runs all of Leo's unit tests from the bridge:
```python
# Add leo-editor to sys.path.
import sys
leo_path = r'C:\leo.repo\leo-editor'
if leo_path not in sys.path:
    sys.path.append(leo_path)
# Start the bridge.
import leo.core.leoBridge as leoBridge
path = r'C:\leo.repo\leo-editor\leo\test\unitTest.leo'
controller = leoBridge.controller(
    gui='nullGui',
    loadPlugins=False,  # True: attempt to load plugins.
    readSettings=True,  # True: read standard settings files.
    silent=False,       # True: don't print signon messages.
    verbose=False)      # True: print informational messages.
# Run the unit tests.
g = controller.globals()
g.app.failFast = False
c = controller.openLeoFile(path)
c.k.simulateCommand('run-all-unit-tests-locally')
    # run-all-unit-tests-locally:
        # With readSettings=False: FAILED (failures=8, errors=1, skipped=33)
        # With readSettings=True:  OK (skipped=33)
    # run-all-unit-tests-externally: OK (skipped=51)

```
I'm not sure why there should be failures when readSettings == False.

**Update**: The various failures look benign.  I'll revise the failing tests so they are skipped without settings:

- `@`test abbrevCommands.next_place requires abbreviation-related settings.
- `@`test g.app.config @buttons and @commands logic clearly require settings.
- `@`test ifplatform probably depends on settings.
- Various tests use k.manufactureKeyPressForCommandName. They must fail without settings.

**Comment by vitalije** [comment]
65ca5ec3 fixes the issue with sqlite3.OperationalErrror
However there are 4 failures and one error while running unit tests.
Neither of those 5 tests seem to be related to cacher instances.

**Comment by vitalije** [comment]
Yes, when I changed readSettings=False to readSettings=True, I have only one failure and it is in the test converting rst to pdf using docutils. The result has type bytes, while test expects str. Nothing to do with the leoCache nor leoBridge.

**Comment by edreamleo** [comment]
@vitalije Many thanks for fixing #658. This is a big step forward.

And thanks for your interest in this issue.  I agree that the test failures do not look significant.

My plan is to skip tests as necessary when readSettings is False.

**Comment by edreamleo** [comment]
Fixed at 776ffde in devel.  Still works with Vitalije's changes to leoBridge.py :-)
